examples/ignite_simple.py

```diff
@@ -0,0 +1,134 @@
+
+(1) Execute this code directly.
+    $ python ignite_simple.py
+
+
+(2) Execute through CLI.
+    $ STUDY_NAME=`optuna create-study --direction maximize --storage sqlite:///example.db`
+    $ optuna study optimize ignite_simple.py objective --n-trials=100 --study $STUDY_NAME \
+      --storage sqlite:///example.db
+"""
+
+from __future__ import division
+from __future__ import print_function
+
+from ignite.engine import create_supervised_evaluator
+from ignite.engine import create_supervised_trainer
+from ignite.engine import Events
+from ignite.metrics import Accuracy
+import torch
+from torch import nn
+import torch.nn.functional as F
+from torch.optim import SGD
+from torch.utils.data import DataLoader
+from torch.utils.data import Subset
+from torchvision.datasets.mnist import MNIST
+from torchvision.transforms import Compose
+from torchvision.transforms import Normalize
+from torchvision.transforms import ToTensor
+
+import optuna
+
+
+EPOCHS = 10
+TRAIN_BATCH_SIZE = 64
+VAL_BATCH_SIZE = 1000
+N_TRAIN_EXAMPLES = 3000
+N_VALID_EXAMPLES = 1000
+
+
+class Net(nn.Module):
+    def __init__(self, trial):
+        # Optimize dropout rate in a convolutional neural network.
+        super(Net, self).__init__()
+        self.conv1 = nn.Conv2d(1, 10, kernel_size=5)
+        self.conv2 = nn.Conv2d(10, 20, kernel_size=5)
+        dropout_rate = trial.suggest_uniform('dropout_rate', 0, 1)
+        self.conv2_drop = nn.Dropout2d(p=dropout_rate)
+        self.fc1 = nn.Linear(320, 50)
+        self.fc2 = nn.Linear(50, 10)
+
+    def forward(self, x):
+        x = F.relu(F.max_pool2d(self.conv1(x), 2))
+        x = F.relu(F.max_pool2d(self.conv2_drop(self.conv2(x)), 2))
+        x = x.view(-1, 320)
+        x = F.relu(self.fc1(x))
+        x = F.dropout(x, training=self.training)
+        x = self.fc2(x)
+        return F.log_softmax(x, dim=-1)
+
+
+def get_data_loaders(train_batch_size, val_batch_size):
+    data_transform = Compose([ToTensor(), Normalize((0.1307,), (0.3081,))])
+
+    train_data = MNIST(download=True, root=".", transform=data_transform, train=True)
+    val_data = MNIST(download=False, root=".", transform=data_transform, train=False)
+
+    train_loader = DataLoader(Subset(train_data, range(N_TRAIN_EXAMPLES)),
+                              batch_size=train_batch_size, shuffle=True)
+    val_loader = DataLoader(Subset(val_data, range(N_VALID_EXAMPLES)),
+                            batch_size=val_batch_size, shuffle=False)
+
+    return train_loader, val_loader
+
+
+def objective(trial):
+    # Create a convolutional neural network.
+    model = Net(trial)
+    device = 'cpu'
+    if torch.cuda.is_available():
+        device = 'cuda'
+
+    # We optimize the learning rate and momentum factor of optimizer.
+    lr = trial.suggest_loguniform('lr', 1e-7, 1e-1)
+    momentum = trial.suggest_uniform('momentum', 0.1, 0.9)
+
+    optimizer = SGD(model.parameters(), lr=lr, momentum=momentum)
```

Comment:
How about suggesting optimizer's type?
(e.g., [chainer_simple.py#L55](https://github.com/pfnet/optuna/blob/master/examples/chainer_simple.py#L55))

Reply:
Sounds good! I'll fix it.

Reply:
Fixed: https://github.com/pfnet/optuna/pull/559/commits/313113b67908df5095d6502b9dfba765214253ae